### ISSUE TITLE
Document replace not supported

### ISSUE THREAD

**Issue by madmaxpt** [bwaldvogel/mongo-java-server#227]
Running the following code :
```java
try(var client = MongoClients.create(connectString))
{
    var collection = client.getDatabase("testDatabase").getCollection("testCollection");
    var replaceOptions = new ReplaceOptions().upsert(true);
    System.out.println(collection.bulkWrite(
            Collections.singletonList(new ReplaceOneModel<>(Filters.eq(new BsonString("myId")),
                    Document.parse("{\"value\": \"test\"}"), replaceOptions))
    ));
    System.out.println(collection.bulkWrite(
            Collections.singletonList(new ReplaceOneModel<>(Filters.eq(new BsonString("myId")),
                    Document.parse("{}"), replaceOptions))
    ));
}
```

Displays the following output :

```
AcknowledgedBulkWriteResult{insertedCount=0, matchedCount=0, removedCount=0, modifiedCount=0, upserts=[BulkWriteUpsert{index=0, id=BsonString{value='myId'}}], inserts=[]}
AcknowledgedBulkWriteResult{insertedCount=0, matchedCount=1, removedCount=0, modifiedCount=0, upserts=[], inserts=[]}
```

The document is matched but not modified.

Running it against a MongoDB server, we get the following output :
```
AcknowledgedBulkWriteResult{insertedCount=0, matchedCount=0, removedCount=0, modifiedCount=0, upserts=[BulkWriteUpsert{index=0, id=BsonString{value='myId'}}], inserts=[]}
AcknowledgedBulkWriteResult{insertedCount=0, matchedCount=1, removedCount=0, modifiedCount=1, upserts=[], inserts=[]}
```

And the document is modified as expected.

Using mongo-java-server version 1.44.0 and mongodb-driver-sync version 4.11.1

**Comment by bwaldvogel** [comment]
Fixed via https://github.com/bwaldvogel/mongo-java-server/pull/229

**Comment by bwaldvogel** [comment]
Fix is released in version `1.45.0`
